sysboundary/outflow.cpp

```diff
@@ -127,6 +89,78 @@ namespace SBC {
+        for(int i=0; i<6; i++) {
+          sP.facesToSkipVlasov[i] = true;
+        }
+
+        std::vector<std::string> thisSpeciesFaceList;
+        if(!Readparameters::get(pop + "_outflow.face", thisSpeciesFaceList)) {
+           if(myRank == MASTER_RANK) cerr << __FILE__ << ":" << __LINE__ << " ERROR: This option has not been added for population " << pop << "!" << endl;
+           exit(1);
+        }
+
+        for(auto& face : thisSpeciesFaceList) {
+          if(face == "x+") { facesToProcess[0] = true; sP.facesToSkipVlasov[0] = false; }
+          if(face == "x-") { facesToProcess[1] = true; sP.facesToSkipVlasov[1] = false; }
+          if(face == "y+") { facesToProcess[2] = true; sP.facesToSkipVlasov[2] = false; }
+          if(face == "y-") { facesToProcess[3] = true; sP.facesToSkipVlasov[3] = false; }
+          if(face == "z+") { facesToProcess[4] = true; sP.facesToSkipVlasov[4] = false; }
+          if(face == "z-") { facesToProcess[5] = true; sP.facesToSkipVlasov[5] = false; }
+        }
+
+        std::array<std::string, 6> vlasovSysBoundarySchemeName;
+        if(!Readparameters::get(pop + "_outflow.vlasovScheme_face_x+", vlasovSysBoundarySchemeName[0])) {
+           if(myRank == MASTER_RANK) cerr << __FILE__ << ":" << __LINE__ << " ERROR: This option has not been added for population " << pop << "!" << endl;
+           exit(1);
+        }
+        if(!Readparameters::get(pop + "_outflow.vlasovScheme_face_x-", vlasovSysBoundarySchemeName[1])) {
+           if(myRank == MASTER_RANK) cerr << __FILE__ << ":" << __LINE__ << " ERROR: This option has not been added for population " << pop << "!" << endl;
+           exit(1);
+        }
+        if(!Readparameters::get(pop + "_outflow.vlasovScheme_face_y+", vlasovSysBoundarySchemeName[2])) {
+           if(myRank == MASTER_RANK) cerr << __FILE__ << ":" << __LINE__ << " ERROR: This option has not been added for population " << pop << "!" << endl;
+           exit(1);
+        }
+        if(!Readparameters::get(pop + "_outflow.vlasovScheme_face_y-", vlasovSysBoundarySchemeName[3])) {
+           if(myRank == MASTER_RANK) cerr << __FILE__ << ":" << __LINE__ << " ERROR: This option has not been added for population " << pop << "!" << endl;
+           exit(1);
+        }
+        if(!Readparameters::get(pop + "_outflow.vlasovScheme_face_z+", vlasovSysBoundarySchemeName[4])) {
+           if(myRank == MASTER_RANK) cerr << __FILE__ << ":" << __LINE__ << " ERROR: This option has not been added for population " << pop << "!" << endl;
+           exit(1);
+        }
+        if(!Readparameters::get(pop + "_outflow.vlasovScheme_face_z-", vlasovSysBoundarySchemeName[5])) {
+           if(myRank == MASTER_RANK) cerr << __FILE__ << ":" << __LINE__ << " ERROR: This option has not been added for population " << pop << "!" << endl;
+           exit(1);
+        }
+        for(uint i=0; i<6 ; i++) {
```

Comment:
@ursg I don't like embedded for loops with the same index... Even though you told me it should work all right...